Release/include/cpprest/http_client.h

```diff
@@ -101,6 +101,7 @@ class http_client_config
         , m_set_user_nativehandle_options([](native_handle)->void{})
 #if !defined(_WIN32) && !defined(__cplusplus_winrt)
         , m_ssl_context_callback([](boost::asio::ssl::context&)->void{})
+        , m_tlsext_host_name(true)
```

Comment:
minor: Can we have the variable name to reflect the type it is representing, bool here?
For eg, something like m_sni_enabled? 
m_tlsext_host_name gives an impression that it is storing the host name to be set.